text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+  print( thirdContext.GetValue("say-something") )  // prints "bar"
+}
+
+func DoWork(){
+  Context::SetValue( "say-something", "bar") 
+}
+```
+
+## Referencing multiple contexts
+If context propagation is automantic, does the user ever need to reference a 
+context object directly? Sometimes. Even when automated context propagation is 
+an available option, there is no restriction which says that concerns must only 
+ever access the current context. 
+
+For example, if a concern wanted to merge the data beween two contexts, at 
+least one of them will not be the current context.
+
+```php
+mergedContext = MergeCorrelations( Context::GetCurrent(), otherContext)
+Context::SetCurrent(mergedContext)
+```
+
+## Falling back to explicit contexts
+Sometimes, suppling an additional version of a function which uses explict 
+contexts is necessary, in order to handle edge cases. For example, in some cases 
+an extracted context is not intended to be set as current context. An 
+alternate extract method can be added to the API to handle this.
+
+```php
+// Most of the time, the extract function operates on the current context.
+Extract(headers)
+
+// When a context needs to be extracted without changing the current 
+// context, fall back to the explicit API.
+otherContext = ExtractWithContext(Context::GetCurrent(), headers)
+```
+
+
+# Internal details
+
+![drawing](img/context_propagation_details.png)
+
+## Default HTTP headers
+OpenTelemetry currently uses two standard header formats for context propagation. 
+Their properties and requirements are integrated into the OpenTelemetry APIs.
+
+**Span Context -** The OpenTelemetry Span API is modeled on the `traceparent` 
+and `tracestate` headers defined in the [W3C Trace Context specification](https://www.w3.org/TR/trace-context/). 
+
+**Correlation Context -** The OpenTelemetry Correlations API is modeled on the 
+`Correlation-Context` headers defined in the [W3C Correlation Context specification](https://w3c.github.io/correlation-context/). 
+
+## Context management and in-process propagation
+
+In order for Context to function, it must always remain bound to the execution 
+of code it represents. By default, this means that the programmer must pass a 
+Context down the call stack as a function parameter. However, many languages 
+provide automated context management facilities, such as thread locals. 
+OpenTelemetry should leverage these facilities when available, in order to 
+provide automatic context management.
+
+## Pre-existing context implementations
+
+In some languages, a single, widely used context implementation exists. In other 
+languages, there many be too many implementations, or none at all. For example, 
+Go has a the `context.Context` object, and widespread conventions for how to 
+pass it down the call stack. Java has MDC, along with several other context 
+implementations, but none are so widely used that their presence can be 
+guranteed or assumed.
```

Comment:
s/guranteed/guaranteed/

Reply:
Thanks for the spellcheck @fbogsany, I beleive I got them all.